# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineSource.h
# Commit: eb376fd954e1
# Full Hash: eb376fd954e1287b8e4f504f743b7159f1b514d0
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Change MediaEngineSource::SetTrack to take a MediaTrack base class. r=padenot
#   
#   This allows us to pass in different sub classes, which the next patch will do.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95933
# ==============================================================================

diff -r 2799805eef3b -r eb376fd954e1 dom/media/webrtc/MediaEngineSource.h
--- a/dom/media/webrtc/MediaEngineSource.h	Thu Nov 05 16:41:45 2020 +0000
+++ b/dom/media/webrtc/MediaEngineSource.h	Thu Nov 05 16:42:07 2020 +0000
@@ -28,7 +28,7 @@
 
 class MediaEnginePhotoCallback;
 class MediaEnginePrefs;
-class SourceMediaTrack;
+class MediaTrack;
 
 /**
  * Callback interface for TakePhoto(). Either PhotoComplete() or PhotoError()
@@ -113,12 +113,12 @@
                             const char** aOutBadConstraint) = 0;
 
   /**
-   * Called by MediaEngine when a SourceMediaTrack has been provided for the
-   * source to feed data to.
+   * Called by MediaEngine when a MediaTrack has been provided for the source to
+   * feed data to.
    *
    * This must be called before Start.
    */
-  virtual void SetTrack(const RefPtr<SourceMediaTrack>& aTrack,
+  virtual void SetTrack(const RefPtr<MediaTrack>& aTrack,
                         const PrincipalHandle& aPrincipal) = 0;
 
   /**